roles/kubernetes-apps/ansible/tasks/nodelocaldns.yml

```diff
@@ -9,10 +9,21 @@
     - { name: nodelocaldns, file: nodelocaldns-deamonset.yml, type: daemonset }
   register: nodelocaldns_manifests
   vars:
-    clusterIP: "{{ skydns_server }}"
+    clusterIP: |-
+      {% if dns_mode in ['kubedns', 'coredns', 'coredns_dual'] %}
+          "{{ skydns_server }}"
+      {% elif dns_mode == 'dnsmasq_kubedns' %}
+          "{{ dnsmasq_dns_server }}"
+      {% elif dns_mode == 'manual' %}
+          "{{ manual_dns_server }}"
+      {% endif %}
     secondaryclusterIP: "{{ skydns_server_secondary }}"
   when:
     - enable_nodelocaldns == True
     - inventory_hostname == groups['kube-master'] | first
   tags:
     - nodelocaldns
+
+- name: Debug templates
+  debug:
+    msg: "{{ lookup('file', '{{ kube_config_dir }}/nodelocaldns-config.yml') }}"
```

Comment:
You might need to add:

```
  when:
    - enable_nodelocaldns == True
    - inventory_hostname == groups['kube-master'] | first
```

So you only debug on the node where the file exists